src/jdk.compiler/share/classes/com/sun/tools/javac/parser/ReferenceParser.java

```diff
@@ -100,106 +109,157 @@ public Reference parse(String sig) throws ParseException {
         Name member;
         List<JCTree> paramTypes;
 
-        Log.DeferredDiagnosticHandler deferredDiagnosticHandler
-                = new Log.DeferredDiagnosticHandler(fac.log);
+        Log.DeferredDiagnosticHandler dh = new Log.DeferredDiagnosticHandler(fac.log);
 
         try {
             int slash = sig.indexOf("/");
-            int hash = sig.indexOf("#", slash + 1);
+            int afterSlash = slash + 1;
+            int hash = sig.indexOf("#", afterSlash);
+            int afterHash = hash + 1;
             int lparen = sig.indexOf("(", Math.max(slash, hash) + 1);
-            if (slash > -1) {
-                moduleName = parseModule(sig.substring(0, slash));
-            } else {
-                moduleName = null;
-            }
-            if (slash > 0 && sig.length() == slash + 1) {
+            int afterLparen = lparen + 1;
+
+            moduleName = switch (slash) {
+                case -1 -> null;
+                case 0 -> throw new ParseException(0, "dc.ref.syntax.error");
```

Comment:
There's a slight change in error output. Although it is likely insignificant, I felt I should note it. For example, compare the error output for `{@link //java.lang.Object}`.

Reply:
Yes, there were some minor changes to a few (3?) tests because of this work. because of the enhanced precision of the positions.

Reply:
In this case, not only the position has changed, but also the error description.

Before:
```
: error: unexpected text
 * {@link //java.lang.Object}
   ^
```
After:
```
error: syntax error in reference
 * {@link //java.lang.Object}
          ^
```